signers/legacy/search_response.go

```diff
@@ -21,41 +22,47 @@ func NewSearchResponseSigner(digest func() hash.Hash) *SearchResponseSigner {
 	}
 }
 
-func (v *SearchResponseSigner) CreateSignable(req *http.Request, authHeaders map[string]string, rw *signers.SignableResponseWriter) []byte {
+func (v *SearchResponseSigner) CreateSignable(body string, nonce string) []byte {
 	var b bytes.Buffer
-	b.WriteString(authHeaders["nonce"])
-	b.WriteString("\n")
-	b.WriteString(req.Header.Get("X-Authorization-Timestamp"))
-	b.WriteString("\n")
-	b.WriteString(rw.Body.String())
+	b.WriteString(nonce)
+	b.WriteString(body)
 	return b.Bytes()
 }
 
 func (v *SearchResponseSigner) SignResponse(req *http.Request, rw *signers.SignableResponseWriter, secret string) (string, *signers.AuthenticationError) {
-	authHeaders := ParseAuthHeadersSearch(req)
-	if _, ok := authHeaders["nonce"]; !ok {
-		return "", signers.Errorf(403, signers.ErrorTypeInvalidAuthHeader, "Nonce must be present in authentication headers.")
+	auth_headers := map[string]string{}
+	auth_fields := []string{
+		"acquia_solr_time",
+		"acquia_solr_nonce",
+		"acquia_solr_hmac",
+	}
+
+	for _, field_name := range auth_fields {
+		auth_cookie, err := req.Cookie(field_name)
+		if err != nil {
+			logger.Print("Error retrieving:", field_name)
+		} else {
+			auth_headers[field_name] = auth_cookie.Value
+		}
 	}
-	if req.Header.Get("X-Authorization-Timestamp") == "" {
-		return "", signers.Errorf(403, signers.ErrorTypeMissingRequiredHeader, "Authorization timestamp for request is required.")
+
+	if _, ok := auth_headers["acquia_solr_nonce"]; !ok {
+		return "", signers.Errorf(403, signers.ErrorTypeInvalidAuthHeader, "Nonce must be present in authentication headers.")
```

Comment:
Why is only the nonce validated here?  Do we need to validate directly or can we rely on the parseAuthHeaders function somehow?